### DISCUSSION TITLE
Proper way of adding custom product types

### DISCUSSION THREAD

**Discussion by Robak08** [saleor/saleor#7652] [Deployment]
Hi, 
I was just wondering about switching to saleor from strapi, and hence my question what is proper (if any) way of extending product types to incorporate custom fields in product schema - will it be through plugin + changes to dashboard?

I'm trying to achieve something along the lines: 
New product type - course (having different sections and steps as repeatable component or something similar (maybe JSON object)), then upon creating/updating order with this product type (with specific payment/order status) there will be userCourses content-type populated with corresponding copy of specific course/courses in order.

Is this approach achievable? If so, what would be proper way of handling.

**Comment by maarcingebala** [comment]
To add custom fields to products you can use attributes. An attribute can represent various types of data: string values, booleans, files, rich-text blocks, numerics. Each product type has a set of product and variant attributes that you can configure. For anything that cannot be handled by attributes, you can also use metadata, which allows you to add any data to products, orders and many more objects in Saleor.

Product types: https://docs.saleor.io/docs/3.0/dashboard/configuration/products
Attributes: https://docs.saleor.io/docs/3.0/dashboard/configuration/attributes
Metadata: https://docs.saleor.io/docs/3.0/dashboard/catalog/products#how-to-manage-metadata-in-the-product-setup-page
